### ISSUE TITLE
An empty API_INTERNAL_URL yields apiBase='' instead of the fallback, because ?? does not catch empty strings

### ISSUE THREAD

**Issue by koeylp** [yuthstyle88/108jobs-clean#112]
## Symptom

`.env.example` ships `API_INTERNAL_URL=` — set, but empty, and deliberately so ("Left empty here on purpose"). Anyone who copies it to `.env` gets an `apiBase` of `""`, not the staging fallback the code appears to promise. Every `/api/*` rewrite then points at a relative path on the Next server itself and 404s.

## Cause

`next.config.ts:265`:

```js
const apiBase = process.env.API_INTERNAL_URL ?? 'https://api-staging.108jobs.com';
```

`??` is nullish coalescing: it catches `null` and `undefined` only. An empty string is neither, so `apiBase = ""` and the rewrite destination collapses:

```js
{ source: '/api/:path*', destination: `${apiBase}/:path*` }
// -> destination: "/v4/files/22/<id>.png"   — a Next route that does not exist
```

The failure looks like a missing page rather than a misconfigured proxy, which is what makes it slow to diagnose: the request never leaves the Next server, so nothing appears in the backend log to correlate against.

## Why the shipped default makes it likely

The comment tells a reader the variable is optional locally and that production throws if it is unset. Both are true. What is not true is the implied "unset locally means fall back to staging" — the shipped empty value is set, so the fallback never applies, and local dev silently proxies nowhere.

## The same repo already handles this correctly elsewhere

api-108heros-dev treats blank env vars as absent on purpose, with the reasoning written down (`crates/infra/src/settings/mod.rs`):

> Comma-separated, like `app_108jobs_CORS_ORIGIN`, and blank-tolerant for the same reason: an unpopulated k8s Secret key arrives as an empty string and must leave the file's value alone rather than emptying the list.

The web config is the fragile version of the same pattern.

## Suggested fix

Treat blank as absent:

```js
const apiBase = process.env.API_INTERNAL_URL?.trim() || 'https://api-staging.108jobs.com';
```

Note this interacts with #111 — whichever value the fallback should be depends on whether `apiBase` is meant to be an origin or to include `/api`. Fixing the empty-string handling without settling #111 would make a local dev environment fall back to staging, which is arguably worse than 404ing loudly.

## Repro

1. `API_INTERNAL_URL=` (as shipped in `.env.example`).
2. Start the dev server, request any proxied path such as `/api/v4/site`.
3. 404 from Next, and nothing in the backend log.

## Notes

Found while verifying the Site Appearance upload end to end.
